Shared-Infrastructure/Initialize-SharedInfrastructureDeployment.ps1

```diff
@@ -0,0 +1,139 @@
+[CmdletBinding()]
+param(
+    [Parameter(Mandatory = $false)]
+    [ValidateSet("DEV", "PP", "PRD", "MO")]
+    [string]$SubscriptionAbbreviation = "DEV",
+    [Parameter(Mandatory = $false)]
+    [ValidateSet("AT", "TEST", "TEST2", "DEMO", "PP", "PRD", "MO")]
+    [string[]]$EnvironmentNames = ($ENV:EnvironmentNames | ConvertFrom-Json),
+    [Parameter(Mandatory = $false)]
+    [ValidateSet("West Europe", "North Europe")]
+    [string]$Location = "West Europe"
+)
+
+$TemplateFilePath = "$PSScriptRoot/templates/subscription.template.json"
+$TemplateParametersFilePath = "$PSScriptRoot/templates/subscription.parameters.json"
+
+try {
+    # --- Are We Logged in?
+    $IsLoggedIn = (Get-AzureRmContext -ErrorAction SilentlyContinue).Account
+    if (!$IsLoggedIn) {
+        throw "You are not logged in. Run Add-AzureRmAccount to continue"
+    }
+
+    # --- Create Resource Groups
+    $ManagementResourceGroupName = "das-$SubscriptionAbbreviation-mgmt-rg".ToLower()
+    $ResourceGroupList = [System.Collections.ArrayList]::new(@($ManagementResourceGroupName))
+    $ResourceGroupList.AddRange(@($EnvironmentNames | ForEach-Object { "das-$_-shared-rg".ToLower() }))
+
+    $ResourceGroupList | ForEach-Object {
+        Write-Host "- Creating Resource Group: $_"
+        $ResourceGroup = Get-AzureRmResourceGroup -Name $_ -Location $Location -ErrorAction SilentlyContinue
+        if (!$ResourceGroup) {
+            $null = New-AzureRmResourceGroup -Name $_ -Location $Location -Confirm:$false
```

Comment:
Should we add a tags parameter?

Reply:
Yes potentially.

The only argument against it is that this will be going over existing environments created with an earlier version of the automation.